OSAT-VideoCompositor/Classes/AVPlayerLayer.swift

```diff
@@ -0,0 +1,87 @@
+//
+//  VideoPlayer.swift
+//  OSAT-VideoCompositor
+//
+//  Created by Rohit Sharma on 13/10/22.
+//
+
+import AVFoundation
+import UIKit
+
+public protocol AVPlayerProtocol: AnyObject {
+    func play()
+    func pause()
+    func set(url: URL)
+}
+
+public typealias AVPlayerCustomView = UIView & AVPlayerProtocol
+
+open class AVPlayerView: AVPlayerCustomView {
+    private var player: AVPlayer?
+    private var playerLayer: AVPlayerLayer?
+    private var url: URL?
+    private var asset: AVAsset?
+    private var avPlayerItem: AVPlayerItem?
+    
+    public override init(frame: CGRect = .zero) {
+        super.init(frame: frame)
+        setVideoPlayer()
+    }
+    
+    public init(frame: CGRect, url: URL?) {
+        self.url = url
+        super.init(frame: frame)
+        
+        setVideoPlayer()
+    }
+    
+    public required init?(coder: NSCoder) {
+        super.init(coder: coder)
+    }
+    
+    // MARK: - Public Apis
+    open override func draw(_ rect: CGRect) {
+        super.draw(rect)
+        playerLayer?.frame = bounds
+    }
+    
+    open func play() {
+        player?.play()
+    }
+    
+    open func pause() {
+        player?.pause()
+    }
+    
+    open func transitionView(to size: CGSize) {
+        frame.size = size
+        playerLayer?.frame.size = size
+    }
+    
+    open func set(url: URL) {
+        self.url = url
+        setVideoPlayer()
+    }
+    
+    // For test
+    open func getPlayer() -> AVPlayer? {
+        return player
+    }
+    // MARK: - Private Methods
+    private func setVideoPlayer() {
```

Comment:
If we want to replace new video, how it will be done?
make generic function which will have automatically accept new video url without reinitiate.

sample function
```
private func setupVideoPlayer() {
        guard let result = self.mtVideoComposition else { return }
        let playerItem = AVPlayerItem.init(asset: result.composition)
        playerItem.videoComposition = result.videoComposition
        if avPlayerLayer == nil {
            avPlayer = AVPlayer.init(playerItem: playerItem)
            avPlayerLayer = AVPlayerLayer.init(player: avPlayer!)
            self.avPlayer?.addPeriodicTimeObserver(
                forInterval: CMTime(seconds: 0.01, preferredTimescale: result.composition.duration.timescale),
                queue: nil,
                using: { (currentTime) in
                    self.currentTime = CGFloat(currentTime.seconds)
                    self.synThumbnailWithPlayer(currentTime: currentTime)
                })
            avPlayerLayer!.frame = self.playerView.bounds
            self.playerView.layer.addSublayer(avPlayerLayer!)
            
            guard let player = self.avPlayer else { return }
            player.seek(to: .zero)
            
        }
        else {
            guard let player = self.avPlayer else { return }
            player.replaceCurrentItem(with: playerItem)
        }
        self.registerNotifications()
    }
```

Reply:
done